### DISCUSSION TITLE
Can WalletConnectSwiftV2 be used for macOS now?

### DISCUSSION THREAD

**Discussion by livid** [WalletConnect/walletconnect-monorepo#1597] [Developer Support]
I am trying to add Sign in with Ethereum to a SwiftUI macOS app.

After the first obstacle:

- Example code below requires Starscream 3 while the latest Starscream package is 4

```
extension WebSocket: WebSocketConnecting { }

struct SocketFactory: WebSocketFactory {
    func create(with url: URL) -> WebSocketConnecting {
        return WebSocket(url: url)
    }
}
```

Now I am stuck in this error when I call `let uri = try await Pair.instance.create()`:

```
Auth token creation error: The operation couldn’t be completed. (WalletConnectKMS.KeychainError error 1.)
```

It seems to be something related to Keychain.

**Comment by llbartekll** [comment]
> Example code below requires Starscream 3 while the latest Starscream package is 4

it is correct, we have found it more stable around iOS versions but you can provide your own implementation if it conforms to the protocol

> Now I am stuck in this error when I call let uri = try await Pair.instance.create():

I have never experienced that problem, do you have your app open sourced so we can check?

**Reply by livid** [reply]
It is not in our main branch yet:

https://github.com/Planetable/Planet

We'll push a simple macOS example app to explore the possibility of integrating with WalletConnectSwiftV2.

**Reply by chadyj** [reply]
Awesome @livid! Can't wait to see a mac app. This will be a first for our Swift project :)

**Reply by livid** [reply]
Thank you for this awesome library! @llbartekll @chadyj 

Our planned usage is to present the QR code as a `.sheet {}` in SwiftUI, and users can scan that code to authenticate. This would be much better than opening a webpage in Chrome and using a URL scheme to jump back to the app.

By the way, I tried Trust, Frontier, and Rainbow, but it seems now no Wallet can do 2.0 Auth yet?

**Reply by kailuo** [reply]
Thank you! @chadyj @llbartekll @livid
We've tried the latest WalletConnectSwiftV2 in a demo project (branch: develop), seems that the error still exists on macOS while it works on iOS:
`Auth token creation error: The operation couldn’t be completed. (WalletConnectKMS.KeychainError error 1.)`

Please refer to the demo project: https://github.com/kailuo/walletdemov2

**Reply by llbartekll** [reply]
I created an issue for this:
https://github.com/WalletConnect/WalletConnectSwiftV2/issues/598

**Reply by livid** [reply]
> Awesome @livid! Can't wait to see a mac app. This will be a first for our Swift project :)

Today we have released a beta version of our app with WalletConnect V1. Here are some screenshots.

[Planet](https://planetable.xyz) is a native macOS app for building, subscribing, and hosting decentralized blogs. For example, you can use it to follow the latest updates from `vitalik.eth` and now tipping a .eth website with WalletConnect like this flow:

<img width="495" alt="connect-wallet" src="https://user-images.githubusercontent.com/11427/205035433-dc97dec5-148f-4bf2-b173-5b3445ba4c01.png">

<img width="1140" alt="qr-code-example" src="https://user-images.githubusercontent.com/11427/205035471-2216dd90-bc41-41ba-b28b-9bb10a9e6cd6.png">

<img width="1140" alt="tip-tiers" src="https://user-images.githubusercontent.com/11427/205035550-8d21232b-f574-4724-bdee-8879c2a1bef9.png">

<img width="1140" alt="sending" src="https://user-images.githubusercontent.com/11427/205036286-80259778-94c1-4659-ad56-cdefbba260fa.png">

<img width="1140" alt="transaction-history" src="https://user-images.githubusercontent.com/11427/205035601-5cbff76b-533d-457e-9008-ce63c225f024.png">

We will start upgrading to V2 once the sample code is working on macOS.

**Reply by chadyj** [reply]
That's awesome @livid!!! Congrats on the launch.

**Comment by chadyj** [comment]
> By the way, I tried Trust, Frontier, and Rainbow, but it seems now no Wallet can do 2.0 Auth yet?

Correct.  Auth just launched, and currently no wallets support it yet.  
However, a few dozen support v2 Sign.  More coming every day!

**Comment by flypaper0** [comment]
@livid @kailuo thanks for responding error, 

I have managed to fix KeychainError with enabling "Keychain Sharing" capability. (With empty app groups list). 

In your [sample app](https://github.com/kailuo/walletdemov2) keychain error was: -34018 errSecMissingEntitlement

**Reply by livid** [reply]
Awesome! Thank you so much! Now we will continue our work on integrating WalletConnectV2.

**Reply by chadyj** [reply]
Great news @livid!
Please share screenshots/links when you get v2 up and running. Good luck!
